tasks/scrapeData/runScraper.js

```diff
@@ -0,0 +1,131 @@
+import * as datetime from '../../lib/datetime.js';
+import * as geography from '../../lib/geography.js';
+
+const numericalValues = ['cases', 'tested', 'recovered', 'deaths', 'active'];
+
+/*
+  Combine location information with the passed data object
+*/
+function addLocationToData(data, location) {
+  Object.assign(data, location);
+
+  delete data.scraper;
+
+  // Store for usage in ratings
+  data._scraperDefinition = location;
+
+  return data;
+}
+
+/*
+  Check if the provided data contains any invalid fields
+*/
+function isValid(data) {
+  if (data.cases === undefined) {
+    throw new Error(`Invalid data: contains no case data`);
+  }
+
+  for (const [prop, value] of Object.entries(data)) {
+    if (value === null) {
+      throw new Error(`Invalid data: ${prop} is null`);
+    }
+    if (Number.isNaN(value)) {
+      throw new Error(`Invalid data: ${prop} is not a number`);
+    }
+  }
+
+  for (const prop of numericalValues) {
+    if (data[prop] !== undefined && typeof data[prop] !== 'number') {
+      throw new Error(`Invalid data: ${prop} is not a number`);
+    }
+  }
+
+  return true;
+}
+
+/*
+  Add output data to the cases array. Input can be either an object or an array
+*/
+function addData(cases, location, result) {
+  if (Array.isArray(result)) {
+    if (result.length === 0) {
+      throw new Error(`Invalid data: scraper for ${geography.getName(location)} returned 0 rows`);
+    }
+    for (const data of result) {
+      if (isValid(data, location)) {
+        cases.push(addLocationToData(data, location));
+      }
+    }
+  } else if (isValid(result, location)) {
+    cases.push(addLocationToData(result, location));
+  }
+}
+
+/*
+  Run the correct scraper for this location
+*/
+export function runScraper(location) {
+  const rejectUnauthorized = location.certValidation === false;
+  if (rejectUnauthorized) {
+    // Important: this prevents SSL from failing
+    process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';
+  }
+  if (typeof location.scraper === 'function') {
+    return location.scraper();
+  }
+  if (rejectUnauthorized) {
+    delete process.env.NODE_TLS_REJECT_UNAUTHORIZED;
+  }
+  if (typeof location.scraper === 'object') {
```

Comment:
lots of JS stuff is an object, could be better to use something more robust eventually (lodash isPlainObject for example)

Reply:
Will do when adding unit tests for this file